R/bayesian.R

```diff
@@ -0,0 +1,311 @@
+#### Functions for doing Bayesian inference on model runs
+
+#' Get historical land use data
+#'
+#' Fetch the historical land use data and filter to the desired regions, years,
+#' and GCAM commodities.
+#'
+#' All of the runs in a single ensemble should have the same regions, years, and
+#' commodities, so generally it will only be necessary to run this function once
+#' at the beginning of the analysis.  The resulting data can be matched to the
+#' model results for each of the models run in order to calculate Bayesian
+#' likelihoods.
+#'
+#' This function also computes the grouped variance for use in subsequent
+#' calculations.
+#'
+#' @section TODO:
+#'
+#' Calculate obsvar as a detrended variance instead of total variance.
+#'
+#' @param regions Regions to keep in filtered data. (Default is use all regions.)
+#' @param years Years to keep in filtered data. (Default is use all years.)
+#' @param commodities GCAM commodities to keep in filtered data (Default is to
+#' use all commodities.)
+#' @return Data frame containing the filtered data.
+#' @export
+get_historical_land_data <- function(regions = NULL, years = NULL,
+                                     commodities = NULL)
+{
+    ## silence notes
+    GCAM_commodity <- variable <- year <- area <- obsvar <-
+        region <- land.type <- obs <- NULL
+
+    FAO_land_history <- gcamland::FAO_land_history
+
+    filter <- rep(TRUE, nrow(FAO_land_history))
```

Comment:
What is this doing?

Reply:
Figuring out which lines to keep.  Start with all `TRUE` and progressively `&` it with various conditions.  The rows that are still `TRUE` at the end get kept.